I've tested this extensively and can confirm that this PR fixes the issue in relation to **Test Case 1.** However, I don't think this PR fixes the issue entirely for **Test Case 2**. It's quite buggy.

**Results**
Emulator API 26-29 (8.2.0.GA): Typing in TextField1 and hitting Return results in Focus on TextField3 while virtual keyboard is still shown. To actually type in TextField3, you must click on TextField3 first. 

Emulator API 26-29 (8.2.1.GA): Typing in TextField1 and hitting Return results in Focus on TextField3, while virtual keyboard is still shown. However, it is not required to click on TextField3 to type into it as in 8.2.0.GA. You can start typing right away upon return/focus of TextField3.

Emulator API 26-29 (this PR): Upon hitting return from TextField1, the keyboard disappears but the focus is now on TextField3. To bring up the keyboard, you must click on TextField3 to start typing.

Note: using `softKeyboardOnFocus: Ti.UI.Android.SOFT_KEYBOARD_SHOW_ON_FOCUS` results in endless warnings in the terminal, and overall reduced performance of the emulator. The errors shown are:
```
[WARN]  IInputConnectionWrapper: getTextBeforeCursor on inactive InputConnection
[WARN]  IInputConnectionWrapper: getSelectedText on inactive InputConnection
[WARN]  IInputConnectionWrapper: getTextAfterCursor on inactive InputConnection
[WARN]  IInputConnectionWrapper: beginBatchEdit on inactive InputConnection
[WARN]  IInputConnectionWrapper: endBatchEdit on inactive InputConnection
```

Device Nexus 5X API 26: Both 8.2.0.GA and 8.2.1.GA function the same. Upon hitting return, you can start typing immediately in TextField3 without having to tap on TextField3 first.

Device Nexus 5X API 26 (this PR): With this PR build, the app fails to launch entirely. Error(s) shown below:

```
[WARN]  W/System.err: java.lang.ClassNotFoundException: Didn't find class "ti.modules.titanium.platform.PlatformModule" on path: DexPathList[[zip file "/data/app/com.axway.timob-QiAVF3Dz4l8PXeBh0MkQig==/base.apk"],nativeLibraryDirectories=[/data/app/com.axway.timob-QiAVF3Dz4l8PXeBh0MkQig==/lib/arm64, /data/app/com.axway.timob-QiAVF3Dz4l8PXeBh0MkQig==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
[WARN]  W/System.err:   at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:93)
[WARN]  W/System.err:   at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
[WARN]  W/System.err:   at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
[WARN]  W/System.err:   at org.appcelerator.kroll.runtime.v8.V8Runtime.nativeRunModule(Native Method)
[WARN]  W/System.err:   at org.appcelerator.kroll.runtime.v8.V8Runtime.doRunModule(V8Runtime.java:168)
[WARN]  W/System.err:   at org.appcelerator.kroll.KrollRuntime.runModule(KrollRuntime.java:225)
[WARN]  W/System.err:   at org.appcelerator.titanium.TiLaunchActivity.loadScript(TiLaunchActivity.java:107)
[WARN]  W/System.err:   at org.appcelerator.titanium.TiRootActivity.loadScript(TiRootActivity.java:480)
[WARN]  W/System.err:   at org.appcelerator.titanium.TiLaunchActivity.onResume(TiLaunchActivity.java:192)
[WARN]  W/System.err:   at org.appcelerator.titanium.TiRootActivity.onResume(TiRootActivity.java:499)
[WARN]  W/System.err:   at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1354)
[WARN]  W/System.err:   at android.app.Activity.performResume(Activity.java:7079)
[WARN]  W/System.err:   at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3620)
[WARN]  W/System.err:   at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3685)
[WARN]  W/System.err:   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2898)
[WARN]  W/System.err:   at android.app.ActivityThread.-wrap11(Unknown Source:0)
[WARN]  W/System.err:   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1593)
[WARN]  W/System.err:   at android.os.Handler.dispatchMessage(Handler.java:105)
[WARN]  W/System.err:   at android.os.Looper.loop(Looper.java:164)
[WARN]  W/System.err:   at android.app.ActivityThread.main(ActivityThread.java:6541)
[WARN]  W/System.err:   at java.lang.reflect.Method.invoke(Native Method)
[WARN]  W/System.err:   at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
[WARN]  W/System.err:   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
[ERROR] KrollProxy: (main) [30,30] Error creating proxy
[ERROR] KrollProxy: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object java.lang.Class.newInstance()' on a null object reference
```

**Test Environment**
```
Studio Ver:     5.1.4.201909061933
SDK Ver:        8.3.0.v20191030085531
OS Ver:         10.14.6
Appc NPM:       4.2.15
Appc CLI:       7.1.1
Daemon Ver:     1.1.3
Ti CLI Ver:     5.2.1
Alloy Ver:      1.14.1
Node Ver:       10.16.3
NPM Ver:        6.12.0
```

--- a/android/titanium/src/java/org/appcelerator/titanium/util/TiUIHelper.java
+++ b/android/titanium/src/java/org/appcelerator/titanium/util/TiUIHelper.java
@@ -1119,7 +1119,9 @@ public class TiUIHelper
 		final InputMethodManager imm =
 			(InputMethodManager) view.getContext().getSystemService(Activity.INPUT_METHOD_SERVICE);
 		if (imm != null) {
-			view.requestFocus();
+			if (!view.hasFocus()) {
+				view.requestFocus();
+			}
 			if (show) {
 				imm.showSoftInput(view, InputMethodManager.SHOW_FORCED);
 			} else {

--- a/android/titanium/src/java/org/appcelerator/titanium/util/TiUIHelper.java
+++ b/android/titanium/src/java/org/appcelerator/titanium/util/TiUIHelper.java
@@ -1121,9 +1121,9 @@ public class TiUIHelper
 		if (imm != null) {
 			view.requestFocus();
 			if (show) {
-				imm.showSoftInput(view, InputMethodManager.SHOW_IMPLICIT);
+				imm.showSoftInput(view, InputMethodManager.SHOW_FORCED);
 			} else {
-				imm.hideSoftInputFromWindow(view.getWindowToken(), InputMethodManager.HIDE_IMPLICIT_ONLY);
+				imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
 			}
 		}
 	}

